pkg/resource/bucket/hook.go

```diff
@@ -0,0 +1,171 @@
+			delta.Add("Spec.Logging.LoggingEnabled.TargetBucket", desired.LoggingEnabled.TargetBucket, latest.LoggingEnabled.TargetBucket)
+		} else {
+			if *desired.LoggingEnabled.TargetBucket != *latest.LoggingEnabled.TargetBucket {
+				delta.Add("Spec.Logging.LoggingEnabled.TargetBucket", desired.LoggingEnabled.TargetBucket, latest.LoggingEnabled.TargetBucket)
+			}
+		}
+
+		if ackcompare.HasNilDifference(desired.LoggingEnabled.TargetPrefix, latest.LoggingEnabled.TargetPrefix) {
+			delta.Add("Spec.Logging.LoggingEnabled.TargetPrefix", desired.LoggingEnabled.TargetPrefix, latest.LoggingEnabled.TargetPrefix)
+		} else {
+			if *desired.LoggingEnabled.TargetPrefix != *latest.LoggingEnabled.TargetPrefix {
+				delta.Add("Spec.Logging.LoggingEnabled.TargetPrefix", desired.LoggingEnabled.TargetPrefix, latest.LoggingEnabled.TargetPrefix)
+			}
+		}
+	}
+
+	// TODO(RedbackThomson): Diff LoggingEnabled.TargetGrants
+	return delta, nil
+}
+
+func (rm *resourceManager) newPutBucketLoggingPayload(
+	r *resource,
+) (*svcsdk.PutBucketLoggingInput, error) {
+	res := &svcsdk.PutBucketLoggingInput{}
+	logging := r.ko.Spec.Logging
+
+	res.SetBucket(*r.ko.Spec.Name)
+
+	if logging != nil {
+		loggingStatus := &svcsdk.BucketLoggingStatus{}
+
+		if logging.LoggingEnabled != nil {
+			loggingEnabled := &svcsdk.LoggingEnabled{}
+
+			if logging.LoggingEnabled.TargetBucket != nil {
+				loggingEnabled.SetTargetBucket(*logging.LoggingEnabled.TargetBucket)
+			}
+			if logging.LoggingEnabled.TargetPrefix != nil {
+				loggingEnabled.SetTargetPrefix(*logging.LoggingEnabled.TargetPrefix)
+			}
+
+			grants := []*svcsdk.TargetGrant{}
+			for _, grant := range logging.LoggingEnabled.TargetGrants {
+				newGrant := &svcsdk.TargetGrant{}
+
+				if grant.Permission != nil {
+					newGrant.SetPermission(*grant.Permission)
+				}
+
+				if grant.Grantee != nil {
+					newGrantee := &svcsdk.Grantee{}
+
+					if grant.Grantee.DisplayName != nil {
+						newGrantee.SetDisplayName(*grant.Grantee.DisplayName)
+					}
+
+					if grant.Grantee.EmailAddress != nil {
+						newGrantee.SetEmailAddress(*grant.Grantee.EmailAddress)
+					}
+
+					if grant.Grantee.ID != nil {
+						newGrantee.SetID(*grant.Grantee.ID)
+					}
+
+					if grant.Grantee.Type != nil {
+						newGrantee.SetType(*grant.Grantee.Type)
+					}
+
+					if grant.Grantee.URI != nil {
+						newGrantee.SetURI(*grant.Grantee.URI)
+					}
+				}
+
+				grants = append(grants, newGrant)
+			}
+			if len(grants) > 0 {
+				loggingEnabled.SetTargetGrants(grants)
+			}
+
+			loggingStatus.SetLoggingEnabled(loggingEnabled)
+		}
+		res.SetBucketLoggingStatus(loggingStatus)
+	}
+
+	return res, nil
+}
+
+func (rm *resourceManager) syncLogging(
+	ctx context.Context,
+	r *resource,
+) (created *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncLogging")
+	defer exit(err)
+	input, err := rm.newPutBucketLoggingPayload(r)
+	if err != nil {
+		return nil, err
+	}
+
+	ko := r.ko.DeepCopy()
```

Comment:
Why is this `DeepCopy` needed? I noticed no changes being made in `ko` object.

Reply:
@vijtrip2 Nick is simply copy/pasting from the normal sdk_update template... nothing really wrong with that... though techincally he should be doing if **below** the call to `rm.sdkapi.PutBucketLogging`